### PR TITLE
add FunctionsBenchmark #1183

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -682,6 +682,7 @@ lazy val scioJmh: Project = Project(
   classDirectory in Jmh := (classDirectory in Test).value,
   dependencyClasspath in Jmh := (dependencyClasspath in Test).value,
   libraryDependencies ++= Seq(
+    "org.hamcrest" % "hamcrest-all" % hamcrestVersion,
```
Comment (regadas) [inline build.sbt]: @nevillelyh is it used? or did I missed something?

File: scio-jmh/src/test/scala/com/spotify/scio/util/FunctionsBenchmark.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import scala.collection.JavaConverters._
+
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+class FunctionsBenchmark {
+
+  type T = Set[Int]
+
+  val input = Lists.newArrayList((1 to 100).map(Set(_)).asJava)
```
Comment (regadas) [inline scio-jmh/src/test/scala/com/spotify/scio/util/FunctionsBenchmark.scala]: @nevillelyh I no this is handy but can we replace this with pure scala/java ... one less dep on guava 😄

Comment (jto) [reply scio-jmh/src/test/scala/com/spotify/scio/util/FunctionsBenchmark.scala]: `new ArrayList((1 to 100).map(Set(_)).asJava)`

File: build.sbt (hunk 1 above)
Comment (nevillelyh) [reply build.sbt]: It's needed by `CombineFnTester`.

**Pull request merged**